web/modules/custom/dbcdk_community/src/Plugin/Block/ProfilesBlock.php

```diff
@@ -208,6 +250,48 @@ public function build() {
   }
 
   /**
+   * Get all quarantined profiles.
+   *
+   * @param array $profiles_filter
+   *   An array containing filter arguments.
+   *
+   * @throws \DBCDK\CommunityServices\ApiException
+   *   Throws API Exception if any of the calls to the services fails.
+   *
+   * @return \DBCDK\CommunityServices\Model\Profile[] $profiles
+   *   An array of Profile objects with an active quarantine.
+   */
+  protected function getQuarantinedProfiles(array $profiles_filter) {
```

Comment:
Methods like this point in the direction of a separate ProfileRepository class which can encapsulate such logic. Perhaps later on though.